subprojects/docs/src/docs/userguide/dep-man/04-modeling-features/modeling_features.adoc

```diff
@@ -280,15 +280,258 @@ Contrary to <<#sec:maven-mapping-to-variants, Maven>>, there is no derivation st
+
+Now the _consumer_ needs to depend on this configuration in order to get the right artifact:
+
+.An explicit configuration dependency
+====
+include::sample[dir="userguide/dependencyManagement/modelingFeatures/crossProjectPublications/simple/groovy",files="consumer/build.gradle[tags=explicit-configuration-dependency]"]
+include::sample[dir="userguide/dependencyManagement/modelingFeatures/crossProjectPublications/simple/kotlin",files="consumer/build.gradle.kts[tags=explicit-configuration-dependency]"]
+====
+
+[WARNING]
+====
+Declaring a dependency on an explicit target configuration is _not recommended_ if you plan to publish the component which has this dependency:
+this would likely lead to broken metadata.
+If you need to publish the component on a remote repository, follow the instructions of the <<#sec:variant-aware-sharing,variant-aware cross publication documentation>>.
+====
+
+In this case, we're adding the dependency to the _instrumentedClasspath_ configuration, which is a _consumer specific configuration_.
+In Gradle terminology, this is called a <<declaring_dependencies.adoc#sub:resolvable-consumable-configs,resolvable configuration>>, which is defined this way:
+
+.Declaring a resolvable configuration on the consumer
+====
+include::sample[dir="userguide/dependencyManagement/modelingFeatures/crossProjectPublications/simple/groovy",files="consumer/build.gradle[tags=resolvable-configuration]"]
+include::sample[dir="userguide/dependencyManagement/modelingFeatures/crossProjectPublications/simple/kotlin",files="consumer/build.gradle.kts[tags=resolvable-configuration]"]
+====
+
+[[sec:variant-aware-sharing]]
+=== Variant-aware sharing of artifacts between projects
+
+In the <<#sec:simple-sharing-artifacts-between-projects, simple sharing solution>>, we defined a configuration on the producer side which serves as an exchange of artifacts between the producer and the consumer.
+However, the consumer has to explicitly tell which configuration it depends on, which is something we want to avoid in _variant aware resolution_.
+In fact, we also <<#understanding-variant-selection, have explained>> that it is possible for a consumer to express requirements using _attributes_ and that the producer should provide the appropriate outgoing variants using attributes too.
+This allows for smarter selection, because using a single dependency declaration, without any explicit target configuration, the consumer may resolve different things.
+The typical example is that using a single dependency declaration `project(":myLib")`, we would either choose the `arm64` or `i386` version of `myLib` depending on the architecture.
+
+To do this, we will add attributes to both the consumer and the producer.
+
+[WARNING]
+====
+The difficulty if this step is understanding that once configurations have attributes, they participate in _variant aware resolution_, which means that they are candidates considered whenever _any_ notation like `project(":myLib")` is used.
```

Comment:
`The difficulty if this step is understanding that...` -> `It is important to understand that...`